br/pkg/lightning/mydump/region.go

```diff
@@ -292,19 +294,28 @@ func MakeSourceFileRegion(
 		return regions, subFileSizes, err
 	}
 
+	fileSize := fi.FileMeta.FileSize
+	rowIDMax := fileSize / divisor
+	// for compressed files, suggest the compress ratio is 1% to calculate the rowIDMax.
+	// set fileSize to INF to make sure compressed files can be read until EOF. Because we can't get the exact size of the compressed files.
+	// TODO: update progress bar calculation for compressed files.
+	if fi.FileMeta.Compression != CompressionNone {
+		rowIDMax = fileSize * 100 / divisor
```

Comment:
I assume we should do some tests on it. It seems this estimate can easily allocate too many row ids and make lightning fail to insert rows that need auto id even when the number of rows is far smaller than the type limit.

Reply:
leave a fixme in https://github.com/pingcap/tidb/pull/39153/commits/46ddf535f8ebbea9c97ec56c16bec6146e701389
I will do some tests and maybe fix this problem in the next PR.